code/game/objects/items/defibrillator.dm

```diff
@@ -128,215 +136,193 @@
 	stack_trace("Powercell deleted while powering the defib, this isn't supposed to happen normally.")
 	set_dcell(null)
 
-/mob/living/proc/get_ghost(bypass_client_check = FALSE)
-	if(client) //Let's call up the correct ghost!
-		return null
-	for(var/mob/dead/observer/ghost AS in GLOB.observer_list)
-		if(!ghost) //Observers hard del often so lets just be safe
-			continue
-		if(isnull(ghost.can_reenter_corpse))
-			continue
-		if(ghost.can_reenter_corpse.resolve() != src)
-			continue
-		if(ghost.client || bypass_client_check)
-			return ghost
-	return null
-
-/mob/living/carbon/human/proc/has_working_organs()
-	var/datum/internal_organ/heart/heart = internal_organs_by_name["heart"]
-
-	if(!heart || heart.organ_status == ORGAN_BROKEN || !has_brain())
-		return FALSE
+/obj/item/defibrillator/attack(mob/living/carbon/human/patient, mob/living/carbon/human/user)
+	defibrillate(patient,user)
 
+///Proc for checking that the defib is ready to operate
+/obj/item/defibrillator/proc/defib_ready(mob/living/carbon/human/patient, mob/living/carbon/human/user)
+	if(!ready)
+		to_chat(user, span_warning("Take the paddles out to continue."))
+		return FALSE
+	if(!ishuman(patient))
+		to_chat(user, span_warning("The instructions on [src] don't mention how to resuscitate that..."))
+		return FALSE
+	if(patient.stat != DEAD)
+		user.visible_message(span_warning("[icon2html(src, viewers(user))] \The [src] buzzes: Patient is not in a valid state. Operation aborted."))
+		return FALSE
+	if(patient.wear_suit && (patient.wear_suit.atom_flags & CONDUCT)) // something conductive on their chest
+		user.visible_message(span_warning("[icon2html(src, viewers(user))] \The [src] buzzes: Patient's chest is obscured. Remove suit or armor and try again."))
```

Comment:
Minor thing, but this is explicitly not a 'is wearing suit slot' check, its checking a specific flag that many (but not all) suits have. So the old message was more accurate to what the actual issue is.